ruby/lib/microservice_logging.rb

```diff
@@ -40,10 +34,28 @@ def method_missing(event)
   ].freeze
 
   def with(scoped_properties)
-    MicroserviceLogger.new(:service_name => @service_name,
-                           :clock => @clock,
-                           :output => @output,
-                           :scoped_properties => scoped_properties)
+    MicroserviceLogger.new(service_name: @service_name,
+                           clock: @clock,
+                           output: @output,
+                           scoped_properties: scoped_properties)
+  end
+
+  private
+
+  def event_loggers
+    @event_loggers ||= Hash[
+      @events.map do |method_name|
```

Comment:
Rubocop issue with this was that the constructor was too long, hence extracting to a method. Once the method is extracted is creating the hash lazily is just more idiomatic.